resource-managers/mesos/src/test/scala/org/apache/spark/deploy/rest/mesos/MesosRestServerSuite.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.deploy.rest.mesos
+
+import javax.servlet.http.HttpServletResponse
+
+import org.scalatest.mockito.MockitoSugar
+
+import org.apache.spark.{SparkConf, SparkFunSuite}
+import org.apache.spark.deploy.TestPrematureExit
+import org.apache.spark.deploy.mesos.MesosDriverDescription
+import org.apache.spark.deploy.rest.{CreateSubmissionRequest, CreateSubmissionResponse, SubmitRestProtocolMessage, SubmitRestProtocolResponse}
+import org.apache.spark.internal.config
+import org.apache.spark.scheduler.cluster.mesos.{MesosClusterPersistenceEngineFactory, MesosClusterScheduler}
+
+class MesosRestServerSuite extends SparkFunSuite
+  with TestPrematureExit with MockitoSugar {
+
+  test("test default driver overhead memory") {
+    testOverheadMemory(new SparkConf(), 2384)
+  }
+
+  test("test configured driver overhead memory") {
+    val conf = new SparkConf()
+    conf.set(config.DRIVER_MEMORY_OVERHEAD.key, "1000")
+    testOverheadMemory(conf, 3000)
+  }
+
```

Comment:
if the overhead mem is set it takes the value as it is, when there is no value set then it takes max (0.10*driver_mem, 384). I have added test for the overhead factor.

Reply:
I was specifically asking about the case where it is explicitly configured with a low values e.g. `conf.set(config.DRIVER_MEMORY_OVERHEAD.key, "10")`

In which case they only get 10MB of overhead memory which I am not sure is the intent of your PR because I think you meant to always set a minimum?

Reply:
Sorry If I did not understand you correctly.

Here the intent of PR is to take the value whatever is set for `config.DRIVER_MEMORY_OVERHEAD.key`, if there is no value set for `config.DRIVER_MEMORY_OVERHEAD.key` then take the max (0.10*driver_mem, 384). 

Having another test with `conf.set(config.DRIVER_MEMORY_OVERHEAD.key, "10")` doesn't make the difference with the existing one `conf.set(config.DRIVER_MEMORY_OVERHEAD.key, "1000")` since it is going to take the value whatever set.

Please let me know if it still doesn't make sense.

Reply:
Basically I was asking does it make sense to allow the user to specify a low value for this configuration i.e. by allowing users to do this are they going to break their applications?

Since it cases where it is not set you automatically calculate a minimum value is there a also a minimum we should enforce regardless of what the user requests?

Also a related point, presumably I could also do something nonsensical like `conf.set(config.DRIVER_MEMORY_OVERHEAD.key, "-1000")` which I think would end up subtracting memory unless the config entry is configured to prevent negative values?

Reply:
Thanks @rvesse for explaining it.

I don't think it is a better way to auto-correct the wrong value if it is explicitly configured by the user instead of failing and letting the user know. And also it makes no-way to configure small value if the users want to have based on their workload.

I see that the same is followed in yarn and k8s deployment modes as well, if we want to change this we may have to change in yarn and k8s also.
Please let me know either way.

Reply:
Provided its consistent with other resource managers then it's likely fine as-is